plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreExecutionContext.java

```diff
@@ -267,14 +267,15 @@ private void setSearchPath(String path) {
         }
     }
 
-    void setSessionRole(final DBRProgressMonitor monitor) throws DBCException {
+    private void setSessionRole(@NotNull DBRProgressMonitor monitor) throws DBCException {
         final String roleName = getDataSource().getContainer().getConnectionConfiguration().getProviderProperty(PostgreConstants.PROP_CHOSEN_ROLE);
         if (CommonUtils.isEmpty(roleName)) {
             return;
         }
         try (JDBCSession session = openSession(monitor, DBCExecutionPurpose.UTIL, "Set active role")) {
             try (JDBCStatement dbStat = session.createStatement()) {
-                dbStat.executeUpdate("SET ROLE " + roleName);
+                String sql = "SET ROLE " + getDataSource().getSQLDialect().getQuotedIdentifier(roleName, false, true);
```

Comment:
Let's use prepared statements here instead 🙂

Reply:
Let's not do that for the following reasons: 

1. In the general case, you can not bind everything using JDBC's `?`. AFAIK, it works 100% reliably only when binding column values;
2. There is no point in defending against SQL injection here;
3. Prepared statements are generally more expensive to create which makes them pointless if they are not reused.

Reply:
Good points, okay.